guppylang/cfg/analysis.py

```diff
@@ -85,83 +85,92 @@ def run(self, bbs: Iterable[BB]) -> Result[T]:
 
 # For live variable analysis, we also store a BB in which a use occurs as evidence of
 # liveness.
-LivenessDomain = dict[str, BB]
+LivenessDomain = dict[P, BB]
 
 
-class LivenessAnalysis(BackwardAnalysis[LivenessDomain]):
+class LivenessAnalysis(BackwardAnalysis[LivenessDomain[P]], Generic[P]):
```

Comment:
nit: Reading https://mypy.readthedocs.io/en/stable/generics.html I think the `Generic[P]` here is not required. If you'd like it for explicitness, ISTR the convention is to put Generic *first*....but now I can't find where that might be doc'd!!